cartographer_ros/cartographer_ros/node_options.cc

```diff
@@ -34,6 +36,10 @@ NodeOptions CreateNodeOptions(
       lua_parameter_dictionary->GetDouble("submap_publish_period_sec");
   options.pose_publish_period_sec =
       lua_parameter_dictionary->GetDouble("pose_publish_period_sec");
+  options.trajectory_publish_period_sec =
+      lua_parameter_dictionary->HasKey("trajectory_publish_period_sec")
```

Comment:
Sure, that's the way I did it at first. But then I thought, why break all other people's existing config files because of this? (Cartographer will exit with an error because of the missing dictionary entry.) But if you think that this should be mandatory in the config file, just revert the last commit (5f48bfae3d0cf14fcad6363450099af451fa9486).